api/resource_groups.go

```diff
@@ -304,6 +268,18 @@ func (group ResourceGroupData) Status() string {
 	return "Disabled"
 }
 
+func (group ResourceGroupData) IsV2Group() bool {
+	if group.Props != nil {
+		return false
+	}
+
+	if group.Query != nil {
+		return true
+	}
+
+	return false
```

Comment:
nit: could reduce this the only condition that returns true is `group.Query != nil`

```suggestion
	return group.Query != nil 
```